src/rust/sqs-executor/src/redis_cache.rs

```diff
@@ -247,6 +343,34 @@ impl Cache for RedisCache {
         res
     }
 
+    #[tracing::instrument(skip(self, cacheables))]
+    async fn get_all<CA>(
+        &mut self,
+        cacheables: Vec<CA>,
+    ) -> Result<Vec<CacheResponse>, Self::CacheErrorT>
+    where
+        CA: Cacheable + Send + Sync + 'static,
+    {
+        let span = tracing::span!(
+            tracing::Level::TRACE,
+            "redis_cache.get_all",
+            address = self.address.as_str(),
+            cacheables_len = cacheables.len(),
+        );
+        let _enter = span.enter();
+
+        let (res, ms) = self._get_all(cacheables).timed().await;
+
+        self.metric_reporter
+            .histogram(
+                "redis_cache.get_all.ms",
+                ms as f64,
+                &[tag("success", res.is_ok())],
+            )
+            .unwrap_or_else(|e| error!("failed to report redis_cache.get_all.ms: {:?}", e));
+        res
+    }
+
```

Comment:
I dig how all the reporting logic is in the trait implementation, while the main logic is delegated to the type. Pretty nice!

It also seems like something ripe for a macro, but that can come later :joy:

Reply:
I had the same thought! Might be a nice addition to the grapl_utils project :P